docs/conftest.py

```diff
@@ -34,7 +35,7 @@ def _docdir(request):
         # Don't apply this fixture to io.rst.  It reads files and doesn't write.
         # Implementation from https://github.com/pytest-dev/pytest/discussions/10437
         if "io.rst" not in request.node.name:
-            old_cwd = os.getcwd()
+            old_cwd = Path.cwd()
```

Comment:
is `os.getcwd()` getting deprecated?

Reply:
No, this is to make to standardized usage of PathLike objects rather than `str`.

(It's fine either way, but to not litter `noqa` all around we could make this change.

Reply:
`pathlib` was introduced in 3.4 and is now pretty feature-complete. Seems a good opportunity to update our approach to paths from the function-oriented manipulation of strings to an object-oriented manipulation of paths. `os.getcwd` isn't deprecated and probably isn't going to be. It's just  the next best thing: superceded.

Reply:
One thing to keep in mind though is that `Path` objects are much heavier than strings. It doesn't make a measurable difference here but in intensive loops, switching to `pathlib` can add a significant overhead.

Reply:
Yeah, and I don't think there are places where we are doing a lot of string/path manipulation inside astropy (maybe I missed something?). A lot of this is indeed aesthetic-y changes. But IMO we should indeed move on to using the obj oriented pathlib way as it make manipulating these path much easier, plus in a typing heavy future of python, `os.PathLike` gives a lot more information than `str`.